### ISSUE TITLE
Can't deploy service to overlay network on windows server 2016

### ISSUE THREAD

**Issue by sify21** [moby/moby#36420]
**Description**
I have a windows server 2016 as a manager node in a swarm. I can't create service running on this node using overlay networks.

**Steps to reproduce the issue:**
1.`docker swarm init` to initialize a swarm
2.create a service using `docker service create --network ingress --endpoint-mode dnsrr --publish mode=host,target=9999,published=9999 my_win_image`

**Describe the results you received:**
HNS failed with error: Element not found

**Describe the results you expected:**
service create succeed

**Additional information you deem important (e.g. issue happens only occasionally):**
I can create the service successfully without `--network ` option, and I can access the published port from another machine, this way the created container runs in `nat` network. However if I assign the service to an overlay network(ingress or any other overlay network I create), above error appears.
Because I'm just doing some experiments, my windows server is not yet activated, don't know whether that matters.

**Output of `Debug-ContainerHost.ps1`:**
Checking for common problems
Container Host OS Product Name: Windows Server 2016 Standard
Container Host OS Build Label: 14393.2097.amd64fre.rs1_release_1.180212-1105
Describing Windows Version and Prerequisites
 [+] Is Windows 10 Anniversary Update or Windows Server 2016 316ms
 [+] Has KB3192366, KB3194496, or later installed if running Windows build 14393 155ms
 [+] Is not a build with blocking issues 136ms
 [+] Has 'Containers' feature installed 10.46s
Describing Docker is installed
 [+] A Docker service is installed - 'Docker' or 'com.Docker.Service'  165ms
 [+] Service is running 132ms
 [+] Docker.exe is in path 2.46s
 [+] Docker is registered in the EventLog service 142ms
Describing User has permissions to use Docker daemon
 [+] docker.exe should not return access denied 137ms
Describing Windows container settings are correct
 [+] Do not have DisableVSmbOplock set to 1 159ms
 [+] Do not have zz values set 138ms
 [+] Do not have FDVDenyWriteAccess set to 1 147ms
Describing The right container base images are installed
 [+] At least one of 'microsoft/windowsservercore' or 'microsoft/nanoserver' should  be installed 397ms
Describing Container network is created
 [+] At least one local container network is available 2.63s
 [+] At least one NAT, Transparent, or L2Bridge Network exists 146ms
 [+] NAT Network's vSwitch is internal 135ms
 [+] A Windows NAT is configured if a Docker NAT network exists 223ms
 [+] Specified Network Gateway IP for NAT network is assigned to Host vNIC 214ms
 [+] NAT Network's internal prefix does not overlap with external IP' 148ms
Showing output from: docker info
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 8
Server Version: 17.12.0-ce
Storage Driver: windowsfilter
 Windows:
Logging Driver: json-file
Plugins:
 Volume: local
 Network: ics l2bridge l2tunnel nat null overlay transparent
 Log: awslogs etwlogs fluentd gelf json-file logentries splunk syslog
Swarm: active
 NodeID: aoe2tof2neo95g7dm22vmu9nn
 Is Manager: true
 ClusterID: iva84p0eklunztxtkq8tld1p9
 Managers: 2
 Nodes: 5
 Orchestration:
  Task History Retention Limit: 5
 Raft:
  Snapshot Interval: 10000
  Number of Old Snapshots to Retain: 0
  Heartbeat Tick: 1
  Election Tick: 3
 Dispatcher:
  Heartbeat Period: 5 seconds
 CA Configuration:
  Expiry Duration: 3 months
  Force Rotate: 0
 Autolock Managers: false
 Root Rotation In Progress: false
 Node Address: 10.33.44.132
 Manager Addresses:
  10.33.44.132:2377
  10.33.44.56:2377
Default Isolation: process
Kernel Version: 10.0 14393 (14393.2097.amd64fre.rs1_release_1.180212-1105)
Operating System: Windows Server 2016 Standard
OSType: windows
Architecture: x86_64
CPUs: 24
Total Memory: 128GiB
Name: win-132
ID: 7WQO:OQCL:2W47:DQRE:RQYG:QT5W:KPAT:53X7:FO54:5PFU:MAP2:V62S
Docker Root Dir: C:\ProgramData\Docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: -1
 Goroutines: 164
 System Time: 2018-02-27T10:24:38.573528+08:00
 EventsListeners: 1
Registry: https://index.docker.io/v1/
Labels:
Experimental: true
Insecure Registries:
 127.0.0.0/8
Registry Mirrors:
 https://registry.docker-cn.com/
Live Restore Enabled: false

Showing output from: docker version
Client:
 Version:       17.12.0-ce
 API version:   1.35
 Go version:    go1.9.2
 Git commit:    c97c6d6
 Built: Wed Dec 27 20:05:22 2017
 OS/Arch:       windows/amd64

Server:
 Engine:
  Version:      17.12.0-ce
  API version:  1.35 (minimum version 1.24)
  Go version:   go1.9.2
  Git commit:   c97c6d6
  Built:        Wed Dec 27 20:15:52 2017
  OS/Arch:      windows/amd64
  Experimental: true

Showing output from: docker network ls
NETWORK ID          NAME                DRIVER              SCOPE
u8z65jcq6ika        ingress             overlay             swarm
96bddda28008        nat                 nat                 local
4e6becdab9ac        none                null                local

**Comment by sify21** [comment]
I just find out the problem. I'm using hyper-v for some linux vms, and I have manully created a virtual switch which binds to the ethernet network adapter, so that I can connet to the outside from within the vms. If I delete the swith that I created, things start to work. In that case I also noticed that a virtual switch called Layered-ethernet which binds to the ethernet network adapter is automatically created in hyper-v.
